oc-chef-pedant/lib/pedant/rspec/matchers.rb

```diff
@@ -95,12 +83,38 @@ def array_matches?(spec, value)
         # Note this is going to be slow for large arrays,
         # though our actual usage shows we don't have any arrays on a
         # scale that would matter.
-        return true if spec.all? { |s| value.include? s }
-        # Our comparisons are more complex than simple value
-        spec.each_with_index  do |newspec, x|
-          return false unless PedantHashComparator.new(newspec, @mode).matches? value[x]
+        return true if spec.all? do |s|
+          value.any? do |v|
+            compare_value(s,v)
+          end
+        end
+        return false
+      end
+
+      def compare_value(spec, value)
+        case spec
+        when String
+          normalize(spec) == normalize(value)
+        when Regexp
+          spec =~ value
+        when Array
+          array_matches?(spec, value)
+        when Hash
+          PedantHashComparator.new(spec, @mode).matches?(value)
+        when Proc then
+          spec.call(value)
+        else
+          spec == value
+        end
+      end
+
+      #
+      def normalize(candidate)
+        if candidate.is_a?(String)
+          candidate.sub(/#{Pedant::Config.pedant_platform.server}/, Pedant::Config.pedant_platform.base_resource_url)
```

Comment:
Is there an advantage of coercing both the expected and configured value at the time of the comparison versus changing the code that generates the expected URL to always be correct:

https://github.com/chef/chef-server/pull/1230/files#diff-c681a9689ad6f141cb970701a07cbbc3R71

One of my worries is that since we coerce both the expected and actual value, it will be hard to catch cases were we are generating the wrong url.